solutions/cf-1398C.mdx

```diff
@@ -21,33 +21,85 @@ using pi = pair<ll, ll>;
 void solve(){
 	int n; cin >> n;
 	vector<int> dp(n+1, 0);
-	for(int i=1; i<=n; i++){
+	for (int i=1; i<=n; i++) {
 		char c; cin >> c;
 		dp[i]=c-'0';
 	}
-	for(int i=1; i<=n; i++){
+	for (int i=1; i<=n; i++) {
 		dp[i]+=dp[i-1];
 	}
 	map<int, ll> sumDist;
-	for(int i=0; i<=n; i++){
+	for (int i=0; i<=n; i++) {
 		sumDist[dp[i]-i]++;
 	}
 	ll goodArrays=0;
-	for(pi p: sumDist){
+	for (pi p: sumDist) {
 		ll f=p.second;
+		// this uses the formula n * (n - 1) / 2, and it gets
+            	// the sum of all the positive integers less than n
 		goodArrays+=f*(f-1)/2;
 	}
 	cout << goodArrays << endl;
 }
 
-int main(){
+int main() {
 	int t; cin >> t;
-	for(int i=0; i<t; i++){
+	for (int i=0; i<t; i++) {
 		solve();
 	}
 }
 ```
 
 </CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class GoodSubarrays {
+	static long solve (int arrayLength, String stringArray) {
+		int[] array = new int[arrayLength + 1];
+		for (int x = 1; x <= arrayLength; x++) {  
+			array[x] = stringArray.charAt(x - 1) - '0';
+		}
+		for (int x = 1; x <= arrayLength; x++) { 
+			array[x] += array[x - 1];
+		}
+		Map<Integer, Long> sumDist = new HashMap<>();
+		for (int x = 0; x <= arrayLength; x++) {  
+			if (sumDist.containsKey(array[x] - x)) {  // already in map
+				// add one on to what is already there
+				sumDist.put(array[x] - x, sumDist.get(array[x] - x) + 1);
+			}
+			else {  // not in the map yet, put 1
+				sumDist.put(array[x] - x, (long) 1);
+			}
+		}
+		long goodArrays = 0;
+		for (int key : sumDist.keySet()) {  // for each key
+			long f = sumDist.get(key);
+			// this uses the formula n * (n - 1) / 2, and it gets
```

Comment:
why not just "sum of all positive integers less than `f`"? same for cpp solution